application/forms/Authentication/LoginForm.php

```diff
@@ -60,6 +64,17 @@ public function createElements(array $formData)
                 'class'         => isset($formData['username']) ? 'autofocus' : ''
             )
         );
+        $this->addElement(
+            'checkbox',
+            'rememberme',
+            [
+                'label' => $this->translate('Stay logged in'),
+            ]
+        );
+        if (! $this->hasDb()) {
+            $this->getElement('rememberme')->setAttrib('disabled', true);
```

Comment:
Maybe show also a title in this case: `You can't stay logged in without a database configuration backend`